src/SonarScanner.MSBuild.PreProcessor/PreprocessorObjectFactory.cs

```diff
@@ -65,6 +65,7 @@ public async Task<ISonarWebServer> CreateSonarWebServer(ProcessedArgs args, IDow
                 if (string.IsNullOrWhiteSpace(args.Organization))
                 {
                     logger.LogError(Resources.ERR_MissingOrganization);
+                    logger.LogWarning(Resources.WARN_DefaultHostUrlChanged);
```

Comment:
This is the only part that we can let the user know in the `begin` step that something might go wrong.
I tested it locally and if you pass organization, it works.

Reply:
If the user will be redirected to SonarCloud instead of SoanarQube, the token will not be correct.

If the token is not correct, this is what is displayed to the user:
```
SonarScanner for MSBuild 7.0
Using the .NET Framework version of the Scanner for MSBuild
Pre-processing started.
Preparing working directories...
18:23:21.58  Updating build integration targets...
18:23:24.848  Downloading from https://api.sonarcloud.io/analysis/version failed. Http status code is Forbidden.
18:24:33.508  Downloading from https://api.sonarcloud.io/analysis/jres?os=windows&arch=x64 failed. Http status code is Forbidden.
18:24:33.591  18:24:33.591  WARNING: JRE Metadata could not be retrieved from analysis/jres?os=windows&arch=x64.
18:24:33.673  Downloading from https://api.sonarcloud.io/analysis/jres?os=windows&arch=x64 failed. Http status code is Forbidden.
18:24:33.744  18:24:33.744  WARNING: JRE Metadata could not be retrieved from analysis/jres?os=windows&arch=x64.
18:24:34.507  Fetching analysis configuration settings...

Unhandled Exception: System.Net.Http.HttpRequestException: Response status code does not indicate success: 401 ().
   at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
   at SonarScanner.MSBuild.PreProcessor.WebClientDownloader.<TryDownloadIfExists>d__5.MoveNext() in D:\src\sonar\sonar-scanner-msbuild\src\SonarScanner.MSBuild.PreProcessor\WebClientDownloader.cs:line 68
...
```

Reply:
Excellent point!